airflow/jobs/local_task_job.py

```diff
@@ -208,6 +208,14 @@ def heartbeat_callback(self, session=None):
                 )
                 raise AirflowException("PID of job runner does not match")
         elif self.task_runner.return_code() is None and hasattr(self.task_runner, 'process'):
+            if ti.state == State.SKIPPED:
+                # A DagRun timeout will cause tasks to be externally marked as skipped.
+                dagrun = ti.dag_run
+                execution_time = (dagrun.end_date or timezone.utcnow()) - dagrun.start_date
+                dagrun_timeout = ti.task.dag.dagrun_timeout
+                self.log.debug(f"dagrun_timeout: {dagrun_timeout}")
```

Comment:
```suggestion
```
I don't think we need it.
Can you now add tests

Reply:
Ah yeah, this was for my own testing and I didn't intend to leave it in - I'll remove it and add some tests.

Reply:
Can you update it :)